### PR TITLE
[CWS][SEC-4020] parse args and envs from the new program stack pages

### PR DESCRIPTION
**Pull request by YoannGh**
This PR changes the way system-probe retrieves the arguments and environment variables of a process from the kernel. It uses the `linux_binprm` kernel structure to read arguments and environment variables strings directly from the new program stack as opposed to using the `execve()` syscall pointer arguments.

Relevant functional tests:
`inv -e security-agent.functional-tests --testflags "-test.v -test.run TestProcessContext"`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/ebpf/c/exec.h [hunk 1]
```diff
@@ -32,6 +33,13 @@ struct bpf_map_def SEC("maps/args_envs_progs") args_envs_progs = {
     .max_entries = 10,
 };
 
+struct bpf_map_def SEC("maps/args_envs_seq_progs") args_envs_seq_progs = {
```
Comment (modernplumbing) [inline pkg/security/ebpf/c/exec.h:36]: can you add a comment about what seq progs are? sequential programs?

File: pkg/security/ebpf/c/exec.h [hunk 2]
```diff
@@ -567,6 +575,148 @@ int kprobe_exit_itimers(struct pt_regs *ctx) {
...
+    };
+
+    int i = 0;
+    int n = 0;
+
+    void *perf_ptr = &buff->value[0];
+
+#pragma unroll
+    for (i = 0; i < MAX_ARRAY_ELEMENT_PER_TAIL_SEQ; i++) {
+        void *ptr = &(buff->value[(event.size + sizeof(n)) & (MAX_STR_BUFF_LEN - MAX_ARRAY_ELEMENT_SIZE - 1)]);
```
Comment (modernplumbing) [inline pkg/security/ebpf/c/exec.h:606]: can we have more descriptive var names than ptr and n?

File: pkg/security/ebpf/c/exec.h [hunk 3]
```diff
@@ -567,6 +575,148 @@ int kprobe_exit_itimers(struct pt_regs *ctx) {
...
+
+        n = bpf_probe_read_str(ptr, MAX_ARRAY_ELEMENT_SIZE, (void *)(args_start + offset));
+        if (n > 0) {
+            n--; // remove trailing 0
+
+            // insert size before the string
+            bpf_probe_read(&(buff->value[event.size&(MAX_STR_BUFF_LEN - MAX_ARRAY_ELEMENT_SIZE - 1)]), sizeof(n), &n);
+
+            int len = n + sizeof(n);
+            if (event.size + len >= MAX_PERF_STR_BUFF_LEN) {
```
Comment (modernplumbing) [inline pkg/security/ebpf/c/exec.h:616]: what does perf mean here?

File: pkg/security/ebpf/c/exec.h [hunk 4]
```diff
@@ -576,7 +726,7 @@ int __attribute__((always_inline)) parse_args_and_env(struct pt_regs *ctx) {
     // call it here before the memory get replaced
     fill_span_context(&syscall->exec.span_context);
 
-    bpf_tail_call_compat(ctx, &args_envs_progs, syscall->exec.next_tail);
+    //bpf_tail_call_compat(ctx, &args_envs_progs, syscall->exec.next_tail);
```
Comment (lebauce) [inline pkg/security/ebpf/c/exec.h:729]: nit: remove comment

Comment (modernplumbing) [inline pkg/security/ebpf/c/exec.h:729]: should we delete this? or at least have a comment about when why we're not deleting this just yet?

File: pkg/security/ebpf/c/exec.h (hunk 3 above)
Comment (YoannGh) [reply pkg/security/ebpf/c/exec.h]: Here perf refers to the eBPF perf map. MAX_PERF_STR_BUFF_LEN is the maximum length for the string buffer that we're sending in the args_envs events through the perf map.

File: pkg/security/ebpf/c/exec.h (hunk 4 above)
Comment (YoannGh) [reply pkg/security/ebpf/c/exec.h]: Just deleted it.

File: pkg/security/probe/constantfetch/fallback.go [hunk 5]
```diff
@@ -676,3 +682,66 @@ func ubuntuAbiVersionCheck(kv *kernel.Version, minAbiPerFlavor map[string]int) b
...
+func getLinuxBinPrmPOffset(kv *kernel.Version) uint64 {
+	offset := uint64(152)
+
+	switch {
+	case kv.Code >= kernel.Kernel5_2:
+		offset = 24
+	case kv.IsRH8Kernel():
+		fallthrough
+	case kv.IsAmazonLinuxKernel() && kv.Code == kernel.Kernel4_14 &&
+		(kv.Code.Patch() == uint8(146) || kv.Code.Patch() == uint8(152) || kv.Code.Patch() == uint8(154) ||
```
[the remark this answers is not in the document]
Comment (YoannGh) [reply pkg/security/probe/constantfetch/fallback.go:695]: I don't expect these patch versions to be tested. I just stumbled upon them through btfhub so I added them with their corresponding fallback offsets.

**Review by lebauce — APPROVED** [review]

**Pull request merged**